android/build.gradle

```diff
@@ -368,16 +366,56 @@ task createNativeDepsDirectories(dependsOn: applyJavaPatches) {
     thirdPartyNdkDir.mkdirs()
 }
 
-task downloadBoost(dependsOn: createNativeDepsDirectories, type: Download) {
+def resolveTaskFactory(String name, String artifactLocalName, File reactNativeAndroidDownloadDir, File reanimatedDownloadDir) {
+    return tasks.create(name: name.startsWith("resolve") ? name : "resolve" + name, dependsOn: createNativeDepsDirectories, type: Copy) {
+        from reactNativeAndroidDownloadDir
+        include artifactLocalName
+        into reanimatedDownloadDir
+
+        onlyIf {
+            // First we check whether the file is already in our download directory
+            if (file("$reanimatedDownloadDir/$artifactLocalName").isFile()) {
+                return false
+            }
+
+            // If it is not the case we check whether it was downloaded by ReactAndroid project
+            def tarFile = Paths.get(reactNativeAndroidDownloadDir.toString(), artifactLocalName).toFile()
+            return tarFile.isFile()
+        }
+    }
+}
+
+Task resolveBoost = resolveTaskFactory("resolveBoost", "boost_${BOOST_VERSION}.tar.gz", reactNativeAndroidDownloadDir, downloadsDir)
+Task resolveDoubleConversion = resolveTaskFactory("resolveDoubleConversion", "double-conversion-${DOUBLE_CONVERSION_VERSION}.tar.gz", reactNativeAndroidDownloadDir, downloadsDir)
+Task resolveFolly = resolveTaskFactory("resolveFolly", "folly-${FOLLY_VERSION}.tar.gz", reactNativeAndroidDownloadDir, downloadsDir)
+Task resolveGlog = resolveTaskFactory("resolveGlog", "glog-${GLOG_VERSION}.tar.gz", reactNativeAndroidDownloadDir, downloadsDir)
+
+def reactNativeAndroidProject = findProject(":ReactAndroid")
```

Comment:
Does `:ReactAndroid` project exist on Paper?

If not, can we wrap this with `if (isNewArchitectureEnabled()) { ... }`?

Reply:
No, it does not exist on Paper. 

I wrote it this way on purpose, so it is "architecture neutral" -- I don't really see a reason to split the code with additional conditionals where it can be shared. Unless your main intention is to emphasise that it is architecture-specific.

Reply:
When it comes to Gradle scripts I think it's always better to be explicit 😄 If it's on purpose, then it's okay, but then some comment here would be valuable.